networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/discovery/discv5/DiscV5Service.java

```diff
@@ -83,6 +91,21 @@ public static DiscoveryService create(
     return Optional.of(discoverySystem.getLocalNodeRecord().asEnr());
   }
 
+  @Override
+  public Optional<String> getDiscoveryAddress() {
+    return Optional.of(advertisedAddr.toString());
```

Comment:
It's really unclear what the result of this is actually meant to be. I've never seen anything that defines how to encode a discovery ENR as a multiaddr correctly which is what this is trying to do.  Best I can see is that we use:
```
final NodeRecord nodeRecord = discoverySystem.getLocalNodeRecord();
if (nodeRecord.getUdpAddress().isEmpty() {
  return Optional.empty();
}
final DiscoveryPeer discoveryPeer = new DiscoveryPeer(((Bytes) nodeRecord.get(EnrField.PKEY_SECP256K1), nodeRecord.getUdpAddress().get(), Optional.empty());

return MultiaddrUtil.fromDiscoveryPeer(discoveryPeer); // But this needs to be a new variant that uses /udp/ not /tcp/
```

Reply:
You just can't use the resulting multiaddr for anything useful because you don't have the public key of the node record and this will actually wind up including a hash of the public key.